src/callbacks_step/save_restart.jl

```diff
@@ -156,5 +160,32 @@ function load_restart_file(semi::AbstractSemidiscretization, restart_file)
     load_restart_file(mesh_equations_solver_cache(semi)..., restart_file)
 end
 
+"""
+    load_controller!(integrator, restart_file::AbstractString)
+
+Load the context information for PID- and PIControllers saved in a `restart_file`.
+"""
+function load_controller!(integrator, restart_file::AbstractString)
+    controller = integrator.opts.controller
+    # Read context information for controller
+    h5open(restart_file, "r") do file
+        # Ensure that necessary information was saved
+        if ("qold" in keys(attributes(file))) &&
+           ("dtpropose" in keys(attributes(file)))
+            integrator.qold = read(attributes(file)["qold"])
+            integrator.dtpropose = read(attributes(file)["dtpropose"])
+            # Accept step to use dtpropose already in the first step
+            integrator.accept_step = true
+            # Reevaluate integrator.fsal_first on the first step
+            integrator.reeval_fsal = true
```

Comment:
I think these two settings do not need to be in the `if` condition but should always be set